### PR TITLE
replace setup > general by twig template

### PR DESCRIPTION
**Pull request by orthagh**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

### CODE REVIEW

**Comment by orthagh** [comment]
New settings page:
image

File: templates/pages/setup/general/general_setup.html.twig [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+    ) }}
+
+    {{ fields.dropdownNumberField(
+        'ajax_limit_count',
+        config['ajax_limit_count'],
+        __("Don't show search engine in dropdowns if the number of items is less than"),
+        inline_field_options|merge({
+            'min': 1,
+            'max': 200,
+            'toadd': [__('Never')],
```
Comment (cedric-anne) [inline templates/pages/setup/general/general_setup.html.twig:162]: Maybe it is more secure to force key value (not tested).
```suggestion
            'toadd': {'0': __('Never')},
```

File: templates/pages/setup/general/general_setup.html.twig [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+            'width': 'auto',
+        })
+    ) }}
+
+    {% set allow_search_view_options = inline_field_options %}
+    {% if not config['allow_search_view'] %}
+        {% set allow_search_view_options = allow_search_view_options|merge({
+            'readonly': true,
+            'disabled': true,
+        }) %}
```
Comment (cedric-anne) [inline templates/pages/setup/general/general_setup.html.twig:194]: I know it was done like this previously, but why ? `Items seen` criteria and global search are disctinct features.

File: templates/pages/setup/general/general_setup.html.twig [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+        inline_field_options
+    ) }}
+
+    {% set lock_options = field_options %}
+    {% if not config['lock_use_lock_item'] %}
+        {% set lock_options = lock_options|merge({
+            'readonly': true,
+            'disabled': true,
+        }) %}
+    {% endif %}
```
Comment (cedric-anne) [inline templates/pages/setup/general/general_setup.html.twig:272]: It forces user to first set 'Yes' , then save to be able to configure the feature. Maybe a small javascript listener could enhance UX.

File: templates/pages/setup/general/general_setup.html.twig (hunk 2 above)
Comment (orthagh) [reply templates/pages/setup/general/general_setup.html.twig:194]: I don't know, i'll remove the condition

**Comment by cconard96** [comment]
I'm thinking the Global Search option is reliant on Items Search option because it does use a hidden Items Seen criteria. The difference is it searches across multiple itemtypes instead of one.

**Comment by cedric-anne** [comment]
> I'm thinking the Global Search option is reliant on Items Search option because it does use a hidden Items Seen criteria. The difference is it searches across multiple itemtypes instead of one.

I pushed a commit to allow global search even when "view" search is not allowed.

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**